src/sgame/sg_buildable.cpp

```diff
@@ -1702,6 +1702,35 @@ itemBuildError_t G_CanBuild( gentity_t *ent, buildable_t buildable, int /*distan
 		level.numBuildablesForRemoval = 0;
 	}
 
+	int max_miners = g_maxMiners.Get();
+	if (max_miners > 0)
+	{
+		gentity_t *tmp = g_entities;
+		int miners = 0;
+		for (int i = 0; i < level.num_entities; i++, tmp++) 
+		{
+			if (tmp->s.eType == entityType_t::ET_BUILDABLE) 
+			{
+				HealthComponent* health = tmp->entity->Get<HealthComponent>();
+				if (tmp->buildableTeam != ent->client->pers.team || !health->Alive())
```

Comment:
Also check for null healthcomponent here to be safe.

Reply:
There's `Entities::IsAlive` for this